### DISCUSSION TITLE
Make a property available to all descendants?

### DISCUSSION THREAD

**Discussion by joeattardi** [lit/lit#3268] [Q&A] [answered]
I am writing a library using Lit. There are some properties that need to be accessed by all elements (an i18n provider, config options, etc). 

I want to avoid if I can passing these all down as properties to each level of the hierarchy (what the React world calls prop drilling). 

Is there any lightweight way I can do this? Something similar to React's context API I guess?

**Comment by bennypowers** [comment, marked as answer]
You might like to try the [context lab](https://github.com/lit/lit/tree/main/packages/labs/context).

**Reply by joeattardi** [reply]
This is exactly what I needed, thank you!
